### PR TITLE
Handle absent log.queue gracefully in rd_kafka_set_log_queue

### PR DESCRIPTION
**Pull request by larry-cdn77**
To address issue #3664

### CODE REVIEW

File: src/rdkafka_queue.c [hunk 1]
```diff
@@ -769,6 +769,10 @@ rd_kafka_queue_t *rd_kafka_queue_get_background(rd_kafka_t *rk) {
 rd_kafka_resp_err_t rd_kafka_set_log_queue(rd_kafka_t *rk,
                                            rd_kafka_queue_t *rkqu) {
         rd_kafka_q_t *rkq;
+
+        if (!rk->rk_logq)
+                return RD_KAFKA_RESP_ERR__INVALID_ARG;
```
Comment (edenhill) [inline src/rdkafka_queue.c:774]: I think NOT_CONFIGURED would be better

Comment (larry-cdn77) [reply src/rdkafka_queue.c]: That sounds better, thanks

File: src/rdkafka.h [hunk 2]
```diff
@@ -3404,7 +3404,8 @@ void rd_kafka_queue_forward(rd_kafka_queue_t *src, rd_kafka_queue_t *dst);
  *
  * @remark librdkafka maintains its own reference to the provided queue.
  *
- * @returns RD_KAFKA_RESP_ERR_NO_ERROR on success or an error code on error.
+ * @returns RD_KAFKA_RESP_ERR_NO_ERROR on success or an error code on error,
+ * eg RD_KAFKA_RESP_ERR__INVALID_ARG when log.queue is not set to true.
```
Comment (edenhill) [inline src/rdkafka.h:3408]: NOT_CONFIGURED

File: tests/0039-event.c [hunk 3]
```diff
@@ -190,7 +191,10 @@ int main_0039_event_log(int argc, char **argv) {
         /* Create kafka instance */
         rk     = test_create_handle(RD_KAFKA_PRODUCER, conf);
         eventq = rd_kafka_queue_get_main(rk);
-        rd_kafka_set_log_queue(rk, eventq);
+        err    = rd_kafka_set_log_queue(rk, eventq);
+
+        if (err != RD_KAFKA_RESP_ERR_NO_ERROR)
```
Comment (edenhill) [inline tests/0039-event.c:196]: nit: `TEST_ASSERT(!err, "set_log_queue failed: %s", rd_kafka_err2str(err));`
or even simpler:  `TEST_CALL_ERR__(rd_kafka_set_log_queue(rk, eventq));`

**Review by edenhill — CHANGES_REQUESTED** [review]

File: tests/0039-event.c (hunk 3 above)
Comment (larry-cdn77) [reply tests/0039-event.c]: Thanks, changed it accordingly

**Review by edenhill — APPROVED** [review]
LGTM!

**Review by edenhill — APPROVED** [review]

**Comment by edenhill** [comment]
Thank you!

**Pull request merged**